dataservice/extensions/flask_indexd.py

```diff
@@ -0,0 +1,218 @@
+        # update fields on the target record's object
+        for prop, v in resp.json().items():
+            if hasattr(record, prop):
+                if prop == 'metadata':
+                    record._metadata = v
+                else:
+                    setattr(record, prop, v)
+
+        return record
+
+    def new(self, record):
+        """
+        Registers a new record in indexd
+
+        Sends post with a body containing info about the file:
+
+          {
+            "file_name": "my_file",
+            "hashes": {
+              "md5": "0b7940593044dff8e74380476b2b27a9"
+            },
+            "size": 123,
+            "urls": [
+              "s3://mybucket/mykey/"
+            ],
+            "metadata": {
+              "acls": "phs000000",
+              "kf_id": "PT_00000000"
+            }
+          }
+
+        A successful response from indexd should contain the following:
+
+          {
+            "baseid": <some-uuid>,
+            "did": <some-uuid",
+            "rev": <first-8-char-of-a-uuid>
+          }
+
+        :param record: A file-like object
+        :returns: Non-error response from indexd
+        :throws: Aborts on non-ok http code returned from indexd
+        """
+        # If running in dev mode, dont call indexd
+        if self.url is None:
+            record.uuid = str(uuid.uuid4())
+            record.latest_did = str(uuid.uuid4())
+            return record
+
+        meta = record._metadata
+
+        req_body = {
+            "file_name": record.file_name,
+            "size": record.size,
+            "form": "object",
+            "hashes": record.hashes,
+            "urls": record.urls,
+            "metadata": meta
+        }
+
+        # Register the file on indexd
+        resp = self.session.post(self.url,
+                                 json=req_body)
+
+        resp.raise_for_status()
+        resp = resp.json()
+
+        # Update the record object with the id fields
+        record.uuid = resp['baseid']
+        record.latest_did = resp['did']
+
+        return record
+
+    def update(self, record):
+        """
+        Updates a record in indexd
+
+        :param record: A file-like object
+        """
+        # If running in dev mode, dont call indexd
+        if self.url is None:
+            record.latest_did = str(uuid.uuid4())
+            return record
+
+        # Fetch rev for the did
+        if record.rev is None:
+            url = self.url + record.latest_did
+            resp = self.session.get(url)
+            self.check_response(resp)
+
+            record.rev = resp.json()['rev']
+
+        req_body = {
+            "form": "object",
+            "file_name": record.file_name,
+            "size": record.size,
+            "hashes": record.hashes,
+            "urls": record.urls,
+            "metadata": record._metadata
+        }
+
+        # Update the file on indexd
+        url = '{}{}?rev={}'.format(self.url, record.latest_did, record.rev)
```

Comment:
Might be that I don't understand the indexd API docs, but I'm wondering which uuid is supposed to be used here in the url. It seems that it should be POST /indexd/{baseid} bc you're creating a new version (new did) for the original record identified by the baseid.

Reply:
Yes, it's confusing. All UUIDs referenced in the docs are the `did`. The `baseid` cannot be used anywhere. I've put in a request to make it so that you can look up documents by `baseid`